File: src/client/classes/api/req_res/parse-kftl-text-request.ts
'use strict'

import { GkillAPIRequest } from '../gkill-api-request'

// メモ帳（KFTL）のテキストを解析だけする要求（Go 側 req_res.ParseKFTLTextRequest）。何も書かない。
// 打鍵のたびに投げて「おかしな行」を塗り、保存の直前に投げて未知タグ・未知板名の確認に使う（ADR-0507）。
export class ParseKFTLTextRequest extends GkillAPIRequest {

    kftl_text: string = ""

    constructor() {
        super()
    }

}

File: src/server/gkill/api/req_res/parse_kftl_text_request.go
package req_res

// ParseKFTLTextRequest はメモ帳（KFTL）のテキストを解析だけする要求。何も書かない。
//
// Web のメモ帳が打鍵のたびに投げて「おかしな行」のピンク表示に使い、保存の直前にも投げて
// 未知タグ・未知板名の確認に使う（ADR-0507）。解析は SubmitKFTLTextRequest と同じ
// prepareRequests を通るので、ここで通った入力が送信で弾かれることは無い。
type ParseKFTLTextRequest struct {
	SessionID  string `json:"session_id"`
	KFTLText   string `json:"kftl_text"`
	LocaleName string `json:"locale_name"`
}
